ffmpeg/nvidia_test.go

```diff
@@ -0,0 +1,398 @@
+// +build nvidia
+
+package ffmpeg
+
+import (
+	"fmt"
+	"os"
+	"testing"
+)
+
+func TestNvidia_Transcoding(t *testing.T) {
+	// Various Nvidia GPU tests for encoding + decoding
+	// XXX what is missing is a way to verify these are *actually* running on GPU!
```

Comment:
If it's encoding in software, then it output always will be bit-by-bit exact? And with hardware it will be different?
So probably we can save hash of output made by software codec and check if it is different?

Reply:
> If it's encoding in software, then it output always will be bit-by-bit exact?

I'm not sure if we can depend on software encoding being bitexact anymore since https://github.com/livepeer/lpms/commit/a12c7a342702837229ede660e1500c1912420b4d

> So probably we can save hash of output made by software codec and check if it is different?

It'd tell us that "this is not encoded with x264" but does not actually ensure it was run on the GPU (or the 'correct' GPU as we select specific GPUs in some other tests).

I'm not sure how practical / feasible it is to check this via unit tests, but would love some other ideas. Parsing `nvidia-smi` output (super racy)?

Reply:
>It'd tell us that "this is not encoded with x264" but does not actually ensure it was run on the GPU (or the 'correct' GPU as we select specific GPUs in some other tests).

We're explicitly asking for `h264_nvenc`. If it isn't available will `avlib` return error? Then we should just believe `avlib` that it used GPU indeed.
I don't see any (reliable) way to ensure that GPU (and which one) was used.

Reply:
> If it isn't available will avlib return error? 

Yes, it should.

> Then we should just believe avlib that it used GPU indeed.

The idea is to catch the possibility where a programming error (in the future) causes the GPU to *not* be used when it should have. (And vice versa.) That'd be a regression, and it's valuable to have tests catch that.

> I don't see any (reliable) way to ensure that GPU (and which one) was used.

Agreed, it's tricky. 🤔